### PR TITLE
[KYUUBI #3441] Change default Spark version to 3.3.1

### PR DESCRIPTION
**Pull request by Yikf**
### _Why are the changes needed?_

Fix https://github.com/apache/incubator-kyuubi/issues/3441;

This pr aims to two points as follow:
- Bump Spark 3.3.0 to Spark 3.3.1 in spark-3.3 profile
- Change default Spark version to 3.3.1

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: extensions/spark/kyuubi-spark-connector-tpcds/src/test/scala/org/apache/kyuubi/spark/connector/tpcds/TPCDSQuerySuite.scala [hunk 1]
```diff
@@ -66,7 +66,7 @@ class TPCDSQuerySuite extends KyuubiFunSuite {
       spark.sql("USE tpcds.tiny")
       queries.map { queryName =>
         val in = Utils.getContextOrKyuubiClassLoader
-          .getResourceAsStream(s"kyuubi/tpcds_3.2/$queryName.sql")
+          .getResourceAsStream(s"kyuubi/tpcds_3.3/$queryName.sql")
```
[the remark this answers is not in the document]
Comment (pan3793) [reply extensions/spark/kyuubi-spark-connector-tpcds/src/test/scala/org/apache/kyuubi/spark/connector/tpcds/TPCDSQuerySuite.scala:69]: "3.2" is not Spark version, it's TPC-DS version
https://tpc.org/TPC_Documents_Current_Versions/pdf/TPC-DS_v3.2.0.pdf

[the remark this answers is not in the document]
Comment (pan3793) [reply extensions/spark/kyuubi-spark-connector-tpcds/src/test/scala/org/apache/kyuubi/spark/connector/tpcds/TPCDSQuerySuite.scala:69]: Update schema to match the lastest Spark version is fine

[the remark this answers is not in the document]
Comment (pan3793) [reply extensions/spark/kyuubi-spark-connector-tpcds/src/test/scala/org/apache/kyuubi/spark/connector/tpcds/TPCDSQuerySuite.scala:69]: Does hash change as well?

**Comment by cfmcgrady** [comment]
FYI: Spark-3.3.1-rc2 is out.
https://www.mail-archive.com/dev@spark.apache.org/msg29460.html

**Comment by pan3793** [comment]
The test failed consistently, could you please take a look?

File: .github/workflows/master.yml [hunk 2]
```diff
@@ -385,10 +385,12 @@ jobs:
           kubectl create serviceaccount spark
           kubectl create clusterrolebinding spark-role --clusterrole=edit --serviceaccount=default:spark --namespace=default
           kubectl get serviceaccount
+      # TODO Support more Spark version
       - name: integration tests
         run: >-
           ./build/mvn ${MVN_OPT} clean install
           -Pflink-provided,hive-provided
+          -Pspark-3.2
```
[the remark this answers is not in the document]
Comment (pan3793) [reply .github/workflows/master.yml:393]: it's tricky, what's the specific issue here?

[the remark this answers is not in the document]
Comment (pan3793) [reply .github/workflows/master.yml:393]: oh sorry, i missed we hardcoded the spark image in code, how about changing it to get from ENV then we can override the in test?

[the remark this answers is not in the document]
Comment (pan3793) [reply .github/workflows/master.yml:393]: We can get the compile Spark version by
```
SPARK_VERSION=$("$MVN" help:evaluate -Dexpression=spark.version $@ 2>/dev/null\
    | grep -v "INFO"\
    | grep -v "WARNING"\
    | tail -n 1)
```

[the remark this answers is not in the document]
Comment (pan3793) [reply .github/workflows/master.yml:393]: make sense, thanks

**Review by pan3793 — APPROVED** [review]
LGTM, it's in good shape now, let's wait for the next RC or GA version, thanks @Yikf

**Comment by pan3793** [comment]
Spark 3.3.1 (RC4) vote passed.
https://www.mail-archive.com/dev@spark.apache.org/msg29565.html

**Comment by pan3793** [comment]
Thanks @Yikf, merging to master! Also thanks @wangyum for releasing Spark 3.3.1

**Comment by Yikf** [comment]
Thanks all.

**Pull request closed without merging**